packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,175 @@
+import {Config} from '@react-native-community/cli-types';
+import {execSync} from 'child_process';
+import {logger, CLIError} from '@react-native-community/cli-tools';
+import chalk from 'chalk';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import transformer from 'hermes-profile-transformer';
+import {findSourcemap, generateSourcemap} from './sourcemapUtils';
+/**
+ * Get the last modified hermes profile
+ * @param packageName
+ */
+function getLatestFile(packageName: string): string {
+  try {
+    const file = execSync(`adb shell run-as ${packageName} ls cache/ -tp | grep -v /$ | egrep '\.cpuprofile' | head -1
+        `);
+    return file.toString().trim();
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+function execSyncWithLog(command: string) {
+  if (logger.isVerbose()) {
+    logger.debug(`${command}\n`);
+  }
+
+  return execSync(command);
+}
+
+/**
+ * Get the package name of the running React Native app
+ * @param config
+ */
+function getPackageName(config: Config) {
+  const androidProject = config.project.android;
+
+  if (!androidProject) {
+    throw new CLIError(`
+  Android project not found. Are you sure this is a React Native project?
+  If your Android files are located in a non-standard location (e.g. not inside \'android\' folder), consider setting
+  \`project.android.sourceDir\` option to point to a new location.
```

Comment:
Note: You can write it like here: https://github.com/react-native-community/cli/blob/a90851a34b96ac0b42b1cc32b4a3f6039134da8c/packages/cli/src/commands/link/link.ts#L63-L66 (indentation on lines 41/42/43). We inline them anyway and in the future, will use dedent or similar when that changes.